### PR TITLE
Add language selection dropdown

### LINKED ISSUE #977: [Bug] hreflang codes contain unrecognised undersores
The Google SEO report for jamulus.io says there is a problem with "unknown language codes" in some links using the "hreflang" property.  This is used in the language switcher,  specifically zh_CN, nb_NO, ko_KR, pt_BR, pt_PT.

It seems that if we can change the UI to show for example "zh-CN" and not "zh_CN", that would fix it.

I don't know how much of a problem this is for displaying content in the relevant Google geographic regions though, but it may be worth fixing.

_PS: Google is probably picking it up from [sitemap.xml](https://github.com/jamulussoftware/jamuluswebsite/blob/release/sitemap.xml) in fact. See my comment below._

### PR DESCRIPTION
**Pull request by ignotus666**
**Short description of changes**
Adds a language selection dropdown to the website. The website also detects what language the user's browser is in and initially displays itself in that language.

Scripts automating the addition of new languages have been updated to support adding the full language name + language code.

It also includes the changes from #988 and #990 so as to avoid conflicts between PRs and include all the changes here.

**Context: Fixes an issue? Related issues**
Fixes #977, #986 and #992

**Status of this Pull Request**
Should be ready pending a review.

**What is missing until this pull request can be merged?**
Would be good if some more people can test it.

IMPORTANT: Weblate stuff should be merged first.

**Does this need translation?**

No.

## Checklist

- [x] I've verified that this Pull Request follows the [general code principles](https://github.com/jamulussoftware/jamulus/blob/main/CONTRIBUTING.md#jamulus-projectsource-code-general-principles)
- [x] I waited some time after this Pull Request was opened and all GitHub checks completed without errors. 
- [x] I'm sure that this Pull Request goes to the correct branch

### CODE REVIEW

**Comment by ignotus666** [comment]
@gilgongo It just dawned on me that #988 was meant to be merged into `release`... We can either wait until 3.11 comes out (when `next-release` gets merged into `release`), or right after this PR gets merged, open another PR against `release` with those changes. I can do that as there is other stuff that needs to go into `release` to enable the automated language addition to work.

File: .github/workflows/add-lang.yml [hunk 1]
```diff
@@ -17,12 +17,14 @@ jobs:
           ISSUE_TITLE: ${{ github.event.issue.title }}
           ISSUE_NUM: ${{ github.event.issue.number }}
         run: |
-          NEW_LANG=$(sed 's/.*\[\([^]]*\)].*/\1/' <<< "$ISSUE_TITLE")
-          if [[ $NEW_LANG =~ ^[a-z]{2}(_[A-Z]{2})?$ ]]; then
-            echo "new_lang=$NEW_LANG" >> $GITHUB_ENV
+          LANG_CODE=$(sed 's/.*\[\([^]]*\)].*/\1/' <<< "$ISSUE_TITLE")
+          FULL_LANG=$(sed 's/.*<\([^>]*\)>.*/\1/' <<< "$ISSUE_TITLE")
+          if [[ $LANG_CODE =~ ^[a-z]{2}(-[A-Z]{2})?$ || $FULL_LANG =~ <[^>]+> ]]; then
```
Comment (ignotus666) [inline .github/workflows/add-lang.yml]: Note to self: the `$FULL_LANG` check is wrong (`$FULL_LANG` itself doesn't have < and > characters). It needs to check if those characters are there with text in between them.

Edit: fixed.

**Comment by pljones** [comment]
Could you squash the commits down to a single commit, please.

If it needs to wait for other PRs, please put this one into draft.

**Comment by gilgongo** [comment]
@ignotus666 I'm fine with waiting if that's simpler.. but is it?

**Comment by pljones** [comment]
> @ignotus666 I'm fine with waiting if that's simpler.. but is it?

We've two tagged bugs - gut feel is they should go in before this.

**Comment by ignotus666** [comment]
#977 and #986 are included in this PR - they can be closed.

@gilgongo after this is merged, the stuff that's for release doesn't necessarily have to wait. In fact it might be better as if by chance someone wants to add a language it won't work otherwise. 

@pljones yes, I'll squash all those commits when I get a chance.

**Comment by ignotus666** [comment]
I've been doing some tests and there are going to be conflicts between this and the Weblate PR.

It's caused by the changes to the po folder names and the changed language codes in the po files. I think it might be best to leave those changes for after the Weblate PR and this one are merged. Mainly to make sure the pending translations get in. If there are conflicts at a later stage for any reason Weblate can be force-reset to whatever state `next-release` is in - but the translations will be already there.

@gilgongo I don't know how you replaced those codes within the files, by hand or with a recursive find/replace, but on Weblate, if you change that string (e.g. "pt_PT" -> "pt-PT") in one file in a given language, that change will propagate to all other files in that language.

So anyway, I'm going to revert those lang code changes for now (it only affects the po folders and files), and wait until the Weblate PR and this one have been merged before making that change.

**Comment by ignotus666** [comment]
The check fails because some config files have the lang code changed while the po files/folders don't. This is to be expected.

**Comment by ignotus666** [comment]
@pljones I'm having a hard time squashing all these commits; I think it's because many of them originate from other repos/branches... I'm going to open a new, clean PR.

**Pull request closed without merging**